subsys/net/l2/openthread/Kconfig

```diff
@@ -313,4 +325,58 @@ config OPENTHREAD_REFERENCE_DEVICE
 	help
 	  Enable Thread Certification reference device support in OpenThread stack
 
+config OPENTHREAD_CHANNEL_MONITOR
+	bool "Enable channel monitor support"
+
+config OPENTHREAD_CHANNEL_MANAGER
+	bool "Enable channel manager support"
+	depends on OPENTHREAD_CHANNEL_MONITOR
+
+config OPENTHREAD_CHILD_SUPERVISION
+	bool "Enable child supervision support"
+
+config OPENTHREAD_COAPS
+	bool "Enable secure coap api support"
+	depends on OPENTHREAD_COAP
+
+config OPENTHREAD_DNS_CLIENT
+	bool "Enable DNS client support"
+
+config OPENTHREAD_EXTERNAL_HEAP
+	bool "Enable external heap support"
+
+config OPENTHREAD_IP6_FRAGM
+	bool "Enable ipv6 fragmentation support"
+
+config OPENTHREAD_LEGACY
+	bool "Enable legacy network support"
+
+config OPENTHREAD_LOG_LEVEL_DYNAMIC
+	bool "Enable dynamic log level control"
+
+config OPENTHREAD_MAC_FILTER
+	bool "Enable mac filter support"
+
+config OPENTHREAD_MTD_NETDIAG
+	bool "Enable TMF network diagnostics on MTDs"
+
+config OPENTHREAD_PLATFORM_NETIF
+	bool "Enable platform netif support"
+
+config OPENTHREAD_SETTINGS_RAM
+	bool "Enable volatile-only storage of settings"
+
+config OPENTHREAD_SNTP_CLIENT
+	bool "Enable SNTP Client support"
+
+config OPENTHREAD_CUSTOM_PARAMETERS
```

Comment:
@rlubos do you think it would be possible to pass OPENTHREAD_ENABLE_NCP through this to fix https://github.com/zephyrproject-rtos/zephyr/issues/26383?

Reply:
I've briefly grepped the OpenThread repo, and it seems that `OPENTHREAD_ENABLE_NCP` is an Autotools option, so I'm afraid it has no use with Cmake build system.

Anyway, it seems that the ncp/rcp libraries are being built by OpenThread unconditionally, so my guess is that all we need is to link the appropriate library in the `zephyr/CMakeLists.txt` file in our OpenThread fork. But I'm not 100% certain on that, I had no time yet to investigate this in-depth, sorry.